src/ORM/Behavior/Translate/ShadowTableStrategy.php

```diff
@@ -516,6 +516,11 @@ protected function rowMapper(CollectionInterface $results, string $locale): Coll
                 if ($translation[$field] !== null) {
                     if ($allowEmpty || $translation[$field] !== '') {
                         $row[$field] = $translation[$field];
+
+                        if ($hydrated) {
+                            /** @var \Cake\Datasource\EntityInterface $row */
+                            $row->setDirty($field, false);
```

Comment:
`$row` can be an array, hence array access is used above, so there needs to be an additional `instanceof` check for the `if` above.

Reply:
$hydrated is set to !is_array($row).